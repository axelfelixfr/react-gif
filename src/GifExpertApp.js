import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

const GifExpertApp = () => {
    // const categories = ['Fun', 'Sad', 'Smile'];
    const [categories, setCategories] = useState(['Fun']);
    
    /*
    const handleAdd = () => {
        // setCategories([...categories, 'Other']);
        setCategories(cats => [...cats, 'Other']);
    }*/

    return(
        <>
            <h2>GifExpertApp</h2>
            <AddCategory setCategories={ setCategories } />
            <hr />
            {/* <button onClick={handleAdd}>Agregar</button> */}
            <ol>
                {
                    categories.map(category => (
                        <GifGrid 
                            key={category} 
                            category={category}
                        />
                    ))
                }
            </ol>
        </>
    );
}

export default GifExpertApp;