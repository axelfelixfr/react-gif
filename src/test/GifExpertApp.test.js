import React from 'react';
import { shallow } from 'enzyme';
import GifExpertApp from './../GifExpertApp';
import '@testing-library/jest-dom'

describe('Pruebas en el componente principal <GifExpertApp />', () => {
    
    test('Se debe cargar correctamente el componente', () => {
        const wrapper = shallow(<GifExpertApp/>);

        expect(wrapper).toMatchSnapshot();
    });


    test('Debe de mostrar una lista de categorÃ­as', () => {
        const categories = ['FUN', 'Smile'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);

    })
    
    
})
