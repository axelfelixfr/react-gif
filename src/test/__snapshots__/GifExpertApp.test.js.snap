// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente principal <GifExpertApp /> Debe de mostrar una lista de categor√≠as 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="FUN"
      key="FUN"
    />
    <GifGrid
      category="Smile"
      key="Smile"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas en el componente principal <GifExpertApp /> Se debe cargar correctamente el componente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
